numba/typing/arraydecl.py

```diff
@@ -317,14 +317,14 @@ def sentry_shape_scalar(ty):
                 return
 
             assert ary.ndim == shape.count
-            return signature(self.resolve_T(ary), shape)
+            return signature(self.resolve_T(ary).copy(layout="A"), shape)
```

Comment:
Just wondering if this leads to pessimised behaviours? The "A" layout is safe but if an input layout is "C" or "F" the transpose layout could be represented by the inverse (i.e. "F" or "C"), assuming contiguous? This leading to better performance as downstream code can more likely optimise for such situations.

Reply:
Result layout depends on user input e.g. if `axes=(0, 1)` passed then layout doesn’t inverse. So, it can be resolved only in runtime and it is difficult to specify layout on type inference stage.

Reply:
Numba does have the ability to generate different code if the arguments are literal values (known at compile time) which would allow the optimized result in cases where it could be inferred.  I forget how literal tuples typing works (@sklam?), but this seems like a good place to use that.

Reply:
We can potentially specialize on the literal value of the `axes` but not now because we do not recognize literal tuple just yet.